src/sentry/models/release.py

```diff
@@ -92,8 +92,56 @@ def _on_post(project, trigger):
         ):
             schedule_invalidate_project_config(project_id=project.id, trigger=trigger)
 
+    def _subscribe_project_to_alert_rule(project, release, trigger):
+        """
+        TODO: potentially enable custom query_extra to be passed on ReleaseProject creation (on release/deploy)
+        """
+        from django.utils import timezone
+
+        from sentry.incidents.logic import subscribe_projects_to_alert_rule
+        from sentry.incidents.models import (
+            AlertRule,
+            AlertRuleActivationConditionType,
+            AlertRuleMonitorType,
+        )
+
+        try:
+            project_alert_rules = AlertRule.filter(project=project)
+            if project_alert_rules.exists():
+                # If we have an AlertRule for the project
+                for par in project_alert_rules:
+                    if (
+                        par.monitor_type == AlertRuleMonitorType.ACTIVATED
+                        and par.activation_conditions
+                        == AlertRuleActivationConditionType.RELEASE_CREATION
+                    ):
+                        query_extra = (
+                            f"release:{release.version} AND event.timestamp:>{timezone.now()}"
+                        )
+                        logger.info(
+                            "Attempt subscribe project to activated alert rule",
+                            extra={
+                                "trigger": trigger,
+                                "query_extra": query_extra,
+                            },
+                        )
+                        subscribe_projects_to_alert_rule(
+                            alert_rule=par, projects=[project], query_extra=query_extra
+                        )
+        except Exception as e:
+            logger.exception(
+                "Failed to subscribe project to activated alert rule",
+                extra={
+                    "trigger": trigger,
+                    "exception": e,
+                },
+            )
+
     def post_save(self, instance, **kwargs):
         self._on_post(project=instance.project, trigger="releaseproject.post_save")
+        self._subscribe_project_to_alert_rule(
```

Comment:
it might be nice to be more explicit about when these are created (wherever the Release is being created) -- that's nice as it will decouple these models, allowing the controllers (MVC) to decide when to subscribe to the alert rule.

Reply:
hmm...

This is tacking on to the `ReleaseProject` creation - which is the only time where this would make sense to subscribe to an Activated AlertRule since this is the _activation condition_ 
the `subscribe_project_to_alert_rule` is already the more explicit MVC helper which i'm making use of here.

we might want to think about making our activation conditions more general though 🤔